assets/src/edit-story/components/library/panes/shared/libraryMoveable.js

```diff
@@ -158,9 +166,15 @@ function LibraryMoveable({
 
     // Position the clone that's being dragged.
     const { offsetX, offsetY } = getTargetOffset();
-    const mediaBox = targetBoxRef.current.getBoundingClientRect();
-    const x1 = mediaBox.left - offsetX;
-    const y1 = mediaBox.top - offsetY;
+    const targetBox = targetBoxRef.current.getBoundingClientRect();
+    let x1 = targetBox.left - offsetX;
+    let y1 = targetBox.top - offsetY;
+    // In case of shapes, the clone is larger than the preview
+    // so we position it to center.
+    if ('shape' === type) {
```

Comment:
Will leave this condition in even though in case of media it just wouldn't change the positioning.
Might adjust this again with Text Sets -- not sure yet what will be needed there.